File: LATIHAN1DPBO2025C2/Java/program/Crud.java
import java.util.ArrayList;
import java.util.Scanner;

// deklarasi kelas crud
public class Crud {

    // atribute private
    private ArrayList<PetShop> list; // list of objek Petshop

    /* konstruktor */
    public Crud() {
        list = new ArrayList<>();
    }

    // method create : membuat objek Petshop
    public void create(String id, String nama_produk, String kategori_produk, String harga_produk) {
        // instansiasi objek Petshop
        PetShop pts = new PetShop(id, nama_produk, kategori_produk, harga_produk);
        // masukan objek ke dalam list
        list.add(pts);
    }

    // method read : menampilkan isi list
    public void read() {
        // menampilkan isi list of objek Petshop
        System.out.println("\nList Produk : \n");
        int i;
        for (i = 0; i < list.size(); i++) {
            System.out
                    .println(Integer.toString(i + 1) + ".| " + list.get(i).getId() + " | "
                            + list.get(i).getNama_produk() + " | " + list.get(i).getKategori_produk() + " | "
                            + list.get(i).getHarga_produk());
        }
    }

    // method update : update objek yang ingin diubah
    public void update(String id) {
        Scanner sc = new Scanner(System.in);

        // deklrasi variabel untuk menampung nama produk, kategori produk, harga produk
        // dari data yang
        // akan diubah
        String nama_produk, kategori_produk, harga_produk;

        // perulangan untuk mencari objek yang akan di ubah berdasarkan id yang
        // diinput
        for (int i = 0; i < list.size(); i++) {
            // apabila id pada objek nya sama dengan id inputan
            if (id.equals(list.get(i).getId())) {
                // input isi variabel nama produk, kategori produk, harga produk dari data yang
                System.out.println('\n' + "Masukan data yang ingin diubah !");
                System.out.println("Nama Produk: ");
                nama_produk = sc.next();
                System.out.println("Kategori Produk: ");
                kategori_produk = sc.next();
                System.out.println("Harga Produk: ");
                harga_produk = sc.next();

                // update data yang ingin diubah dengan menggunakan method setter di kelas
                // petshop
                list.get(i).setNama_produk(nama_produk);
                list.get(i).setKategori_produk(kategori_produk);
                list.get(i).setHarga_produk(harga_produk);
                break;
            }
        }
    }

    // method delete : delete objek yang ingin dihapus
    public void delete(String id) {
        // perulangan untuk mencari objek yang di akan di hapus berdasarkan id yang
        // diinput
        for (int i = 0; i < list.size(); i++) {
            // apabila id pada objek nya sama dengan id inputan
            if (id.equals(list.get(i).getId())) {
                list.remove(list.get(i)); // hapus objek pada list
                break;
            }
        }
    }
}

File: LATIHAN1DPBO2025C2/CPP/program/Crud.cpp
// import library dan file
#include <iostream>
#include <list>
#include <string>
#include "Petshop.cpp"

// menggunakan standard namespace
using namespace std;

// deklarasi kelas CRUD
class Crud
{
private:
    list<Petshop> llist; // list of objek Petshop

public:
    // Konstruktor
    Crud() {}

    // Method create
    void create(string id, string nama_produk, string kategori_produk, string harga_produk)
    {
        llist.push_back(Petshop(id, nama_produk, kategori_produk, harga_produk));
    }

    // Method read
    void read()
    {
        cout << "\nList Produk Petshop: " << '\n';
        int i = 0;
        for (list<Petshop>::iterator it = llist.begin(); it != llist.end(); it++)
        {
            cout << (i + 1) << ".| "
                 << it->get_id() << " | "
                 << it->get_nama_produk() << " | "
                 << it->get_kategori_produk() << " | "
                 << it->get_harga_produk() << '\n';
            i++;
        }
    }

    // Method update
    void update(string id)
    {
        string nama_produk, kategori_produk, harga_produk;

        for (list<Petshop>::iterator it = llist.begin(); it != llist.end(); it++)
        {
            if (id == it->get_id())
            {
                cout << "\nMasukkan data baru:\n";
                cout << "Nama Produk: ";
                cin >> nama_produk;
                cout << "Kategori Produk: ";
                cin >> kategori_produk;
                cout << "Harga Produk: ";
                cin >> harga_produk;

                // Memanggil method setter yang sesuai
                it->set_nama_produk(nama_produk);
                it->set_kategori_produk(kategori_produk);
                it->set_harga_produk(harga_produk);
                break;
            }
        }
    }

    // Method delete
    void del(string id)
    {
        for (list<Petshop>::iterator it = llist.begin(); it != llist.end(); it++)
        {
            if (id == it->get_id())
            {
                llist.erase(it);
                break;
            }
        }
    }

    // Destructor
    ~Crud() {}
};
